bin/adhoc-scripts/createPileupObjects.py

```diff
@@ -0,0 +1,179 @@
+    """
+    for doc in pileupDocs:
+        if pileupName == doc["pileupName"]:
+            logger.warning("Reusing pileup configuration for pileup: %s", pileupName)
+            return doc
+    pileupDoc = {
+        'pileupName': "",
+        'pileupType': "",
+        'expectedRSEs': [],
+        'campaigns': [],
+        'active': False}
+    return pileupDoc
+
+
+def parseCampaigns(campDocs, logger):
+    """
+    Parse campaigns from WMCore and convert them into
+    Pileup documents, where:
+     * campaigns: is a union of all campaigns using a given pileup.
+     * expectedRSEs: is a union of all RSEs across different campaigns.
+    """
+    pileupDocs = []
+    for camp in campDocs:
+        # for each secondary dataset, create one pileup document
+        for puName, puRSEs in camp.get("Secondaries", {}).items():
+            puDoc = getPUSchema(puName, pileupDocs, logger)
+            puType = "premix" if puName.split("/")[-1] == "PREMIX" else "classic"
+            puDoc["pileupName"] = puName
+            puDoc["pileupType"] = puType
+            puDoc["expectedRSEs"].extend(puRSEs)
+            puDoc["campaigns"].append(camp["CampaignName"])
+            # make some of these values unique
+            puDoc["expectedRSEs"] = list(set(puDoc["expectedRSEs"]))
+            puDoc["campaigns"] = list(set(puDoc["campaigns"]))
+            pileupDocs.append(puDoc)
+    logger.info("Created %d pileup docs out of %d campaigns.", len(pileupDocs), len(campDocs))
+    return pileupDocs
+
+
+def writePileupDocs(mspileupUrl, puDocs, certDict, logger):
+    """
+    Create pileup documents in MSPileup
+    :param mspileupUrl: string with the MSPileup url
+    :param puDocs: list with pileup documents
+    :param certDict: dictionary with cert and key reference
+    :param logger: logger object
+    :return: None
+    """
+    url = mspileupUrl + "/ms-pileup/data/pileup"
+    mgr = RequestHandler()
+    headers = {'Content-Type': 'application/json'}
+    for doc in puDocs:
+        logger.info("Injecting against %s document: %s", url, doc)
+        resp = mgr.getdata(url, doc, headers, verb='POST', encode=True, decode=True,
+                           ckey=certDict['key'], cert=certDict['cert'])
+
+        logger.debug("Response: %s", resp)
+        if resp and resp.get("result", []):
+            msg = f"Failed to inject pileup document for {doc['pileupName']}. Error: {resp}"
+            logger.critical(msg)
+
+
+def main():
+    """Executes everything"""
+    optmgr = OptionParser()
+    opts = optmgr.parser.parse_args()
+    logger = logging.getLogger('createPileupObjects')
+    logger.setLevel(logging.INFO)
+    logging.basicConfig()
+
+    # setup proxy/cert
+    cert = os.getenv('X509_USER_CERT', '')
+    key = os.getenv('X509_USER_KEY', '')
+    proxy = os.getenv('X509_USER_PROXY', '')
+    if proxy and not cert:
+        cert = proxy
+        key = proxy
+    hdict = {'cert': cert, 'key': key, 'pycurl': True}
```

Comment:
You should add protection that neither proxy or cert/key is set, i.e. user has no X509 environment at all. So far the code assumes that proxy will be set if cert is not provided which is wrong.

Reply:
updating it in a bit.